.github/workflows/test.yml

```diff
@@ -226,14 +226,60 @@ jobs:
           name: generated_binding-${{ matrix.config.os }}-R-${{ matrix.config.r }}-rust-${{ matrix.config.rust-version }}-${{ matrix.config.target || 'default'}}
           path: generated_bindings
 
-      # Run tests again using different bindings
-      - name: Run tests on precomputed bindings shipped with libR-sys
-        if: matrix.config.skip-tests != 'true'
-        run: |
-          . ./ci-cargo.ps1
-          ci-cargo test -vv $(if ($env:RUST_TARGET -ne '') {"--target=$env:RUST_TARGET"} ) '--' --nocapture -ActionName "Running tests for target: $env:RUST_TARGET"
-        env:
-          RUST_TARGET: ${{ matrix.config.target }}
+      # # Run tests again using different bindings
+      # - name: Run tests on precomputed bindings shipped with libR-sys
+      #   if: matrix.config.skip-tests != 'true'
+      #   run: |
+      #     . ./ci-cargo.ps1
+      #     ci-cargo test -vv $(if ($env:RUST_TARGET -ne '') {"--target=$env:RUST_TARGET"} ) '--' --nocapture -ActionName "Running tests for target: $env:RUST_TARGET"
+      #   env:
+      #     RUST_TARGET: ${{ matrix.config.target }}
+  check_generate_bindings_flag:
+    name: Check if [generate bindings] is in latest commit message
+    runs-on: ubuntu-latest
+    outputs:
+      head_commit_message: ${{ steps.get_head_commit_message.outputs.HEAD_COMMIT_MESSAGE }}
+      generate_bindings: ${{ contains(steps.get_head_commit_message.outputs.HEAD_COMMIT_MESSAGE, '[generate bindings]') }}
```

Comment:
This is what we have done all the time. We use GHA to generate all bindings, and then we commit those to the repo. We don't generate the bindings locally and upload them.
There is no build right, because we don't actually build something here, we merely generate the bindings in the `build.rs` script. In fact, I believe #189 is about moving the binding-generation to an `xtask` here.